datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -322,7 +322,8 @@ def _request(self, method, url, options):
             if persist:
                 return getattr(self.session, method)(url, **new_options)
             else:
-                return getattr(requests, method)(url, **new_options)
+                session = self._create_session()
```

Comment:
This looks like a good improvement, especially given your point that we will be using the same objects whether we persist or not.

Will there be any performance impact though?  From my recollection, requests sessions were basically just config wrappers, so impact should be minimal?

Reply:
> Will there be any performance impact though?

Nope, not that I can see, since `requests.get()` etc basically do what we now do explicitly:

https://github.com/psf/requests/blob/967a05bfffcb68f97296eda197b062221c2ebc0d/requests/api.py#L60-L61